### YamlMime:ManagedReference
items:
- uid: System.Threading.ThreadStateException
  id: ThreadStateException
  children:
  - System.Threading.ThreadStateException.#ctor
  - System.Threading.ThreadStateException.#ctor(System.String)
  - System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ThreadStateException
  nameWithType: ThreadStateException
  fullName: System.Threading.ThreadStateException
  type: Class
  summary: "Unntak forårsakes når en <xref href=&quot;System.Threading.Thread&quot;> </xref> har et ugyldig <xref:System.Threading.Thread.ThreadState*>for metodekallet.</xref:System.Threading.Thread.ThreadState*>"
  remarks: "Når en tråd er opprettet, er i minst én av de <xref:System.Threading.ThreadState>angir til det avsluttes.</xref:System.Threading.ThreadState> `ThreadStateException`forårsakes av metoder som ikke kan utføre den forespurte operasjonen på grunn av gjeldende status for en tråd. For eksempel forsøk på å starte en avbrutt tråd ved å kalle <xref:System.Threading.Thread.Start%2A>oppstår en ThreadStateException på en tråd som er avsluttet.</xref:System.Threading.Thread.Start%2A>       ThreadStateException bruker HRESULT COR_E_THREADSTATE, som har verdien 0x80131520.       En liste med opprinnelige verdier for en forekomst av ThreadStateException, kan du se den <xref:System.Threading.ThreadStateException.%23ctor%2A>konstruktører.</xref:System.Threading.ThreadStateException.%23ctor%2A>"
  example:
  - "The following example demonstrates an error that causes the system to throw a `ThreadStateException`.  \n  \n [!code-cpp[ThreadStEx#1](~/add/codesnippet/cpp/t-system.threading.threa_3_1.cpp)]\n [!code-vb[ThreadStEx#1](~/add/codesnippet/visualbasic/t-system.threading.threa_3_1.vb)]\n [!code-cs[ThreadStEx#1](~/add/codesnippet/csharp/t-system.threading.threa_3_1.cs)]  \n  \n This code produces the following output:  \n  \n```  \nIn main. Attempting to restart myThread.  \n    Working thread...  \n    Caught: Thread is running or terminated. Cannot restart.  \n```"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ThreadStateException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException()
  nameWithType: ThreadStateException.ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Initialiserer en ny forekomst av den <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref> klasse standardegenskapene."
  remarks: "Denne konstruktøren initialiserer den `Message` for den nye forekomsten til en system-angitt melding som beskriver feilen, slik som &quot;den forespurte operasjonen ikke kan utføres på tråd på grunn av den gjeldende tilstanden.&quot; Meldingen tar hensyn til gjeldende system kultur.       Tabellen nedenfor viser første egenskapsverdiene for en forekomst av <xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>      | Egenskapen | Verdien |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| Lokaliserte feil melding strengen. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException ();
    parameters: []
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(String)
  nameWithType: ThreadStateException.ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Initialiserer en ny forekomst av den <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref> klassen med en angitt feilmelding."
  remarks: "Denne konstruktøren initialiserer den `Message` for ny forekomst med den `message` parameteren.       Tabellen nedenfor viser første egenskapsverdiene for en forekomst av <xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>      | Egenskapen | Verdien |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| `null`.|   | <xref:System.Exception.Message%2A>| Feil melding strengen. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Feilmelding som forklarer hvorfor unntaket."
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(SerializationInfo,StreamingContext)
  nameWithType: ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Initialiserer en ny forekomst av den <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref> klasse serialiserte dataene."
  remarks: "Denne konstruktøren kalles under deserialisering til reconstitute unntak objektet overføres via en dataflyt. Hvis du vil ha mer informasjon, se [XML og SOAP-serialisering](~/add/includes/ajax-current-ext-md.md)."
  syntax:
    content: protected ThreadStateException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "Den <xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref> som inneholder den serialiserte objektdata om som oppstod et unntak."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "Den <xref:System.Runtime.Serialization.StreamingContext>som inneholder kontekstavhengig informasjon om kilden eller målet.</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(String,Exception)
  nameWithType: ThreadStateException.ThreadStateException(String,Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "Initialiserer en ny forekomst av den <xref href=&quot;System.Threading.ThreadStateException&quot;> </xref> klassen med en angitt feilmelding og en referanse til det interne unntaket er årsaken til dette unntaket."
  remarks: "Et unntak som er registrert som en direkte resultatet av et tidligere unntak må inkludere en referanse til det tidligere unntaket i den <xref:System.Exception.InnerException%2A>egenskapen.</xref:System.Exception.InnerException%2A> Den <xref:System.Exception.InnerException%2A>returnerer samme verdi som er sendt til konstruktøren, eller `null` hvis den <xref:System.Exception.InnerException%2A>egenskapen leverer ikke internt unntak verdien til konstruktøren.</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       Tabellen nedenfor viser første egenskapsverdiene for en forekomst av <xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>      | Egenskapen | Verdien |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Referansen for det indre unntaket. |   | <xref:System.Exception.Message%2A>| Feil melding strengen. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "Feilmelding som forklarer hvorfor unntaket."
    - id: innerException
      type: System.Exception
      description: "Unntaket er årsaken til den gjeldende unntaket. Hvis den `innerException` -parameteren er ikke <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>, oppstår det gjeldende unntaket i en <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> blokken som behandler det interne unntaket."
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Threading.ThreadStateException.#ctor
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException()
  nameWithType: ThreadStateException.ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(String)
  nameWithType: ThreadStateException.ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(SerializationInfo,StreamingContext)
  nameWithType: ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(String,Exception)
  nameWithType: ThreadStateException.ThreadStateException(String,Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Threading.ThreadStateException.#ctor*
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException
  nameWithType: ThreadStateException.ThreadStateException
